js/src/@types/shims.d.ts

```diff
@@ -0,0 +1,17 @@
+import 'flarum/forum/ForumApplication';
+import 'flarum/common/models/User';
+
+declare module 'flarum/forum/ForumApplication' {
+  export default interface ForumApplication {
+    fof_oauth_linkingInProgress: boolean | undefined;
+    fof_oauth_linkingProvider: string | undefined;
+    fof_oauth_loginInProgress: boolean | undefined;
```

Comment:
Just a nitpick, but I think I prefer

```suggestion
    fof_oauth_linkingInProgress?: boolean;
    fof_oauth_linkingProvider?: string;
    fof_oauth_loginInProgress?: boolean;
```